pkg/scheduler/internal/queue/pod_backoff.go

```diff
@@ -17,103 +17,44 @@ limitations under the License.
 package queue
 
 import (
-	"sync"
 	"time"
 
-	ktypes "k8s.io/apimachinery/pkg/types"
-	"k8s.io/kubernetes/pkg/scheduler/util"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
 )
 
-// PodBackoffMap is a structure that stores backoff related information for pods
-type PodBackoffMap struct {
-	// lock for performing actions on this PodBackoffMap
-	lock  sync.RWMutex
-	clock util.Clock
+// PodBackoff is a structure that stores backoff related information for pods
+type PodBackoff struct {
 	// initial backoff duration
 	initialDuration time.Duration
 	// maximal backoff duration
 	maxDuration time.Duration
-	// map for pod -> number of attempts for this pod
-	podAttempts map[ktypes.NamespacedName]int
-	// map for pod -> lastUpdateTime pod of this pod
-	podLastUpdateTime map[ktypes.NamespacedName]time.Time
 }
 
-// NewPodBackoffMap creates a PodBackoffMap with initial duration and max duration.
-func NewPodBackoffMap(initialDuration, maxDuration time.Duration, clock util.Clock) *PodBackoffMap {
-	return &PodBackoffMap{
-		clock:             clock,
-		initialDuration:   initialDuration,
-		maxDuration:       maxDuration,
-		podAttempts:       make(map[ktypes.NamespacedName]int),
-		podLastUpdateTime: make(map[ktypes.NamespacedName]time.Time),
+// NewPodBackoff creates a PodBackoff with initial duration and max duration.
+func NewPodBackoff(initialDuration, maxDuration time.Duration) *PodBackoff {
+	return &PodBackoff{
+		initialDuration: initialDuration,
+		maxDuration:     maxDuration,
 	}
 }
 
-// GetBackoffTime returns the time that nsPod completes backoff
-func (pbm *PodBackoffMap) GetBackoffTime(nsPod ktypes.NamespacedName) (time.Time, bool) {
-	pbm.lock.RLock()
-	defer pbm.lock.RUnlock()
-	if _, found := pbm.podAttempts[nsPod]; found == false {
-		return time.Time{}, false
-	}
-	lastUpdateTime := pbm.podLastUpdateTime[nsPod]
-	backoffDuration := pbm.calculateBackoffDuration(nsPod)
+// GetBackoffTime returns the time that podInfo completes backoff
+func (pbm *PodBackoff) GetBackoffTime(podInfo *framework.PodInfo) time.Time {
+	lastUpdateTime := podInfo.Timestamp
```

Comment:
nit: Use short names for local variables. `last` and `duration`, for example. However, you can just use `podInfo.Timestamp` directly. It's just used once.

Reply:
Done